src/ChunkedCSV.jl

```diff
@@ -203,7 +232,7 @@ function setup_parser(
     0 < nworkers < 256 || throw(ArgumentError("`nworkers` argument must be larger than 0 and smaller than 256."))
     0 <= skipto <= typemax(Int) || throw(ArgumentError("`skipto` argument must be positive and smaller than 9_223_372_036_854_775_808."))
     (header isa Integer && !(0 <= header <= typemax(Int))) && throw(ArgumentError("`header` row number must be positive and smaller than 9_223_372_036_854_775_808."))
-    0 <= limit <= typemax(Int) || throw(ArgumentError("`limit` argument must be positive and smaller than 9_223_372_036_854_775_808."))
+    -1 <= limit <= typemax(Int) || throw(ArgumentError("`limit` argument must be positive and smaller than 9_223_372_036_854_775_808."))
```

Comment:
why did this change? 

should the comment change too?
```suggestion
    -1 <= limit <= typemax(Int) || throw(ArgumentError("`limit` argument must be non-negative and smaller than 9_223_372_036_854_775_808."))
```

Reply:
Ah, good catch -- CSV and ChunkedCSV handled differently what limit=0 means... and this is something I forgot to clean up after I reverted my attempts to comply with CSV.jl.